e2e/cypress/integration/team_settings/join_closed_team_with_not_allowed_email_spec.js

```diff
@@ -0,0 +1,79 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+// ***************************************************************
+// - [#] indicates a test step (e.g. # Go to a page)
+// - [*] indicates an assertion (e.g. * Check the title)
+// - Use element ID when selecting an element. Create one if none.
+// ***************************************************************
+
+// Group: @team_settings
+
+import {getRandomId} from '../../utils';
+import * as TIMEOUTS from '../../fixtures/timeouts';
+
+describe('Team Settings', () => {
+    const randomId = getRandomId();
+    let testTeam;
+
+    before(() => {
+        cy.apiInitSetup().then(({team}) => {
+            testTeam = team;
+            cy.apiAdminLogin();
+            cy.visit(`/${testTeam.name}`);
+        });
+    });
+
+    it('MM-T387 - Try to join a closed team from a NON-mattermost email address via "Get Team Invite Link" while "Allow only users with a specific email domain to join this team" set to "mattermost.com"', () => {
+        // Open 'Team Settings' modal
+        cy.get('.sidebar-header-dropdown__icon').click();
+        cy.findByText('Team Settings').should('be.visible').click();
+
+        // * Check that the 'Team Settings' modal was opened
+        cy.get('#teamSettingsModal').should('exist').within(() => {
+            // # Click on the 'Allow only users with a specific email domain to join this team' edit button
+            cy.get('#allowed_domainsEdit').should('be.visible').click();
+
+            // # Set 'sample.mattermost.com' as the only allowed email domain and save
+            cy.wait(TIMEOUTS.HALF_SEC);
+            cy.focused().type('sample.mattermost.com');
+            cy.findByText('Save').should('be.visible').click();
+
+            // # Close the modal
+            cy.get('#teamSettingsModalLabel').find('button').should('be.visible').click();
+        });
+
+        // # Open the 'Invite People' full screen modal and get the invite url
+        cy.get('.sidebar-header-dropdown__icon').click();
+        cy.get('#invitePeople').find('button').eq(0).click();
+        cy.get('.share-link-input').invoke('val').then((val) => {
+            const inviteLink = val;
+
+            // # Logout from admin account and visit the invite url
+            cy.apiLogout();
+            cy.visit(inviteLink);
+        });
+
+        const email = `user${randomId}@sample.gmail.com`;
+        const username = `user${randomId}`;
+        const password = 'passwd';
+        const errorMessage = 'Email must be from a specific domain (e.g. @example.com). Please ask your team or system administrator for details.';
```

Comment:
It's failing here. I updated the test case to reflect the correct error message.
```suggestion
        const errorMessage = 'The following email addresses do not belong to an accepted domain: sample.mattermost.com. Please contact your System Administrator for details.';
```

(Other non-blocking item but might be good to declare a variable for email domain `sample.mattermost.com`.)

Reply:
Hi @saturninoabril , I tried changing the error message but now my test is failing because of this - 

<img width="1164" alt="Screen Shot 2020-08-16 at 1 38 09 PM" src="https://user-images.githubusercontent.com/44601888/90326666-0b0ad780-dfc6-11ea-85ca-c1694d213d3e.png">

Reply:
Do you know why we are getting different error messages?

Reply:
Looks like that's the old message. Could you try to update your mattermost-server to latest master?

Reply:
Thanks Saturn!